### PR TITLE
DAH-1430 habitat pricing table

### PR DESCRIPTION
**Pull request by kramduckner**
[DAH-1430](https://sfgovdt.jira.com/browse/DAH-1430)

[DAH-1430]: https://sfgovdt.jira.com/browse/DAH-1430?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/javascript/modules/listingDetails/ListingDetailsPricingTable.tsx [hunk 1]
```diff
@@ -200,6 +193,60 @@ const buildAccordions = (
...
+      return chart.numOfHousehold === i
+    })
+
+    const maxOccupancyChart = maxAmiChartsValues.find((chart) => {
+      return chart.numOfHousehold === i
+    })
+
+    if (minOccupancyChart && maxOccupancyChart) {
+      habitatStringArray.push(
+        `${i} people household: $${minOccupancyChart?.amount?.toLocaleString()} to $${maxOccupancyChart?.amount?.toLocaleString()}`
```
Comment (chadbrokaw-exygy) [inline app/javascript/modules/listingDetails/ListingDetailsPricingTable.tsx:229]: Will `i` ever be 1 or 0? If that is possible, then we will need to have a singular and plural string variant. E.g. "1 person household" vs "2 People Household"

Also, will this need to be run through translation?

File: app/javascript/contexts/listingDetails/listingDetailsReducer.ts [hunk 2]
```diff
@@ -29,7 +29,10 @@ const ListingDetailsReducer = createReducer({ units: [] } as ListingDetailsState
       units,
     }
   },
-  [ListingDetailsActions.FinishFetchingAmiCharts]: (state, { payload: amiCharts }) => {
+  [ListingDetailsActions.FinishFetchingAmiCharts]: (
+    state,
+    { payload: { amiCharts, chartsToFetch } }
```
Comment (chadbrokaw-exygy) [inline app/javascript/contexts/listingDetails/listingDetailsReducer.ts:34]: What are `chartsToFetch`?

File: app/javascript/contexts/listingDetails/listingDetailsReducer.ts [hunk 3]
```diff
@@ -38,6 +41,9 @@ const ListingDetailsReducer = createReducer({ units: [] } as ListingDetailsState
         return {
           ...amiChart,
           chartType: amiChart?.values[0]?.chartType,
+          derivedFrom: chartsToFetch.find((chart: RailsAmiChartMetaData) => {
+            return chart.percent === Number(amiChart.percent)
```
Comment (cade-exygy) [inline app/javascript/contexts/listingDetails/listingDetailsReducer.ts:45]: Nit: Any concerns with `amiChart.percent` not being defined?

**Review by chadbrokaw-exygy — APPROVED** [review]
Looks great! Only one question around strings

**Review by cade-exygy — APPROVED** [review]
Testing criteria passes. All looks good based on my understanding, just wondering similar things that @chadbrokaw-exygy  mentioned.

**Pull request merged**